File: app/src/main/java/sk/styk/martin/apkanalyzer/business/task/AppListFromPackageNamesLoader.java
package sk.styk.martin.apkanalyzer.business.task;

import android.content.Context;

import java.util.List;

import sk.styk.martin.apkanalyzer.business.service.launcher.AppBasicDataService;
import sk.styk.martin.apkanalyzer.model.list.AppListData;

/**
 * Loader async task for item for AppListDialog
 * <p>
 * @author Martin Styk
 * @version 05.01.2018.
 */
public class AppListFromPackageNamesLoader extends ApkAnalyzerAbstractAsyncLoader<List<AppListData>> {

    public static final int ID = 5;
    private final AppBasicDataService appBasicDataService;

    private List<String> packageNames;

    public AppListFromPackageNamesLoader(Context context, List<String> packageNames) {
        super(context);
        this.packageNames = packageNames;
        appBasicDataService = new AppBasicDataService(context.getPackageManager());
    }

    @Override
    public List<AppListData> loadInBackground() {
        return appBasicDataService.getForPackageNames(packageNames);
    }

}

File: app/src/main/java/sk/styk/martin/apkanalyzer/business/task/AppListFromPackageNamesLoader.kt
package sk.styk.martin.apkanalyzer.business.task

import android.content.Context

import sk.styk.martin.apkanalyzer.business.service.launcher.AppBasicDataService
import sk.styk.martin.apkanalyzer.model.list.AppListData

/**
 * Loader async task for item for AppListDialog
 *
 * @author Martin Styk
 * @version 05.01.2018.
 */
class AppListFromPackageNamesLoader(context: Context, private val packageNames: List<String>)
    : ApkAnalyzerAbstractAsyncLoader<List<AppListData>>(context) {

    private val appBasicDataService = AppBasicDataService(context.packageManager)

    override fun loadInBackground(): List<AppListData> {
        return appBasicDataService.getForPackageNames(packageNames)
    }

    companion object {
        const val ID = 5
    }

}

